sdk/metric/meter.go

```diff
@@ -200,6 +251,14 @@ func (m *meter) Float64ObservableGauge(name string, options ...metric.Float64Obs
 	return inst, nil
 }
 
+func validateInstrumentName(name string) error {
+	if !instrumentNameRe.MatchString(name) {
+		global.Warn("Invalid Instrument name. Instrument names must consist of 63 or fewer characters including alphanumeric, _, ., -, and start with a letter")
+		return errors.New("invalid instrument name")
```

Comment:
Logging an error and returning the error is redundant. If an error is returned, the user can choose to log it if they want.

Only one of these error handling methods should be used.

Reply:
My intent here was to provide additional information (not in the error) to help folks figure out why the name is invalid.

Reply:
That information will be decoupled from the returned error. If we want to log that should be the way this is handled, but if we want to return an error it should contain the information needed to understand itself.